forest/daemon/src/daemon.rs

```diff
@@ -418,15 +420,18 @@ async fn maybe_fetch_snapshot(
 
 /// Last resort in case a snapshot is needed. If it is not to be downloaded, this method fails and
 /// exits the process.
-fn prompt_snapshot_or_die(config: &Config) -> anyhow::Result<bool> {
+async fn prompt_snapshot_or_die(config: &Config) -> anyhow::Result<bool> {
     if config.client.snapshot_path.is_some() {
         return Ok(false);
     }
     let should_download = if !config.client.auto_download_snapshot && atty::is(atty::Stream::Stdin)
     {
+        // TODO: get the size of the snapshot.
+        let required_size: u64 = snapshot_fetch_size(&config).await.unwrap();
+        let required_size = to_size_string(&required_size.to_bigint().unwrap()).unwrap();
```

Comment:
Let's not use `unwrap`s but propagate the error.

Reply:
Done

Reply:
I meant all of the `unwrap`s. :)